File: 1259-handshakes-that-dont-cross/1259-handshakes-that-dont-cross.py
class Solution:
    def numberOfWays(self, numPeople: int) -> int:
        m = 1000000007
        dp = [0] * (numPeople//2 + 1)
        dp[0] = 1
        for i in range(1, numPeople//2 + 1):
            for j in range(0, i):
                dp[i] += dp[j] * dp[i-j-1] % m
                dp[i] %= m
        return dp[numPeople//2]

File: 1259-handshakes-that-dont-cross/1259-handshakes-that-dont-cross.rb
# @param {Integer} num_people
# @return {Integer}
def number_of_ways(num_people)
    m = 1000000007
    dp = Array.new((num_people/2)+1, 0)
    dp[0] = 1
    (1..(num_people / 2)).each do |i|
      (0...i).each do |j|
        dp[i] += dp[j]*dp[i-j-1]%m
        dp[i] %= m
      end
    end
    return dp[num_people / 2]
end